File: apps/ruby-sdk/lib/firecrawl/models/agent_response.rb
# frozen_string_literal: true

module Firecrawl
  module Models
    # Response from starting an agent task.
    class AgentResponse
      attr_reader :success, :id, :error

      def initialize(data)
        @success = data["success"]
        @id = data["id"]
        @error = data["error"]
      end

      def to_s
        "AgentResponse{id=#{id}, success=#{success}}"
      end
    end
  end
end

File: apps/php-sdk/src/Models/AgentResponse.php
<?php

declare(strict_types=1);

namespace Firecrawl\Models;

final class AgentResponse
{
    public function __construct(
        private readonly bool $success = false,
        private readonly ?string $id = null,
        private readonly ?string $error = null,
    ) {}

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data): self
    {
        return new self(
            success: (bool) ($data['success'] ?? false),
            id: $data['id'] ?? null,
            error: $data['error'] ?? null,
        );
    }

    public function isSuccess(): bool
    {
        return $this->success;
    }

    public function getId(): ?string
    {
        return $this->id;
    }

    public function getError(): ?string
    {
        return $this->error;
    }
}
